db/sql/00_msar.sql

```diff
@@ -282,7 +213,7 @@ $$ LANGUAGE SQL RETURNS NULL ON NULL INPUT;
 
 CREATE OR REPLACE FUNCTION
 msar.get_column_name(rel_id oid, col_id integer) RETURNS text AS $$/*
-Return the name for a given column in a given relation (e.g., table).
+Return the UNQUOTED name for a given column in a given relation (e.g., table).
```

Comment:
I think we should move this to `__msar` because we use it much more frequently than similar functions like `get_column_names`, `get_constraint_name`, or `get_relation_name_or_null`. Forgetting to quote it is much more likely.

Reply:
> because we use it much more frequently

This gives me the impression that you'd like to quote names which are frequently used and not quote names which are not frequently used. Am I interpreting your rationale correctly?

If so, I really strongly disapprove of this line of thinking. In the developer-facing documentation within this PR, I wrote:

> **If it _can_ be unquoted, then it _should_ be unquoted.**

I like this rubric because it has low ambiguity and high conformance with industry best-practices concerning security.

I'm okay with functions within `__msar` operating on quoted names because (as of of PR) this schema is effectively deprecated. It only exists as a matter of technical debt. And after our _full_ quoting refactor (sometime next year) it won't exist at all, with its contained functions and types being refactored into `msar` with unquoted names. I don't want to put things in `__msar` for now just because they're convenient or because we might forget to quote things. I'd rather put as much as we can into `msar` (with unquoted names) so that we can work incrementally towards chipping off at this technical debt.

Does this make sense? I'm worried that I'm not understanding something your comment. Happy to chat about this on a call to clarify.

Reply:
I don't have a strong opinion on this. Initially, I was under the impression that you were open to moving these functions to the `__msar` namespace since you had mentioned this: 

> For each such function we could either: move it into the __msar schema; or we could remove the quoting on the return value (refactoring call sites as needed). We could decide between these two strategies on a per-function basis depending on which strategy appears easiest.

However, I do think I understand your rationale behind keeping it in `msar`, the rationale being adopting the escape-on-output pattern. But, in this specific case, it doesn't seem like we are passing the values of the function(i.e. the column name) around and we are immediately quoting it once we make the call to the function. Therefore, I thought moving it to `__msar` might make sense.

I do have some questions about how we'll go on to remove the `__msar` schema from our codebase, things like what would happen to `__msar.exec_ddl`, what would the future refactor for functions like `__msar.get_column_names`, `__msar.get_relation_name` would look like, etc. but probably this isn't the right place to discuss those.

Reply:
> it doesn't seem like we are passing the values of the function(i.e. the column name) around and we are immediately quoting it

Correct. And because this PR removes the quoting from the `get_column_name` return value, it will allow us to incrementally refactor those call sites in order to continue gradually pushing that quoting down deeper towards the SQL queries whenever we're working with that code.

I did move some functions into `__msar` in this PR, but only because it would have been an unreasonable amount of work to change fully unquote their return values. For example, the function `get_duplicate_col_defs` (previously in `msar`, now in `__msar`) returns `__msar.col_def[]`. That _type_ contains quoted values. Ugh! We'll need to refactor that type and all its usages in order for `get_duplicate_col_defs` to fit cleanly into `msar`. That's a lot of work.

And yes, there will be a bunch of other small decisions to make when it comes time to do this whole refactor. I'll be keen to provide guidance on those decisions. For example:

- For `exec_ddl`, I want that function to go away! It should just be inlined. I feel pretty strongly about this. I've already talked to Brent about this. I can explain more if needed when the time comes.
- For `get_column_names`, it's only used in order to construct `con_def` instances, so it will be straightforward to refactor in conjunction with refactoring the type. It's _work_. But it won't require a ton of decision or redesign.